### PR TITLE
allow unknown fields in proxy config

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/42684

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/context.go [hunk 1]
```diff
@@ -473,7 +473,11 @@ func (s *NodeMetaProxyConfig) MarshalJSON() ([]byte, error) {
 
 func (s *NodeMetaProxyConfig) UnmarshalJSON(data []byte) error {
 	pc := (*meshconfig.ProxyConfig)(s)
-	return jsonpb.Unmarshal(bytes.NewReader(data), pc)
+	// TODO(ramaraochavali): Remove this in 1.18. This is temporarily enabled to allow old
+	// proxies connect to Istiod with removed "ParentShutdownDuration".
```
Comment (howardjohn) [inline pilot/pkg/model/context.go:477]: we should probably keep this forever for safety?

Comment (howardjohn) [reply pilot/pkg/model/context.go:477]: We do that for Istio CRDs

Comment (ramaraochavali) [reply pilot/pkg/model/context.go]: Sure

**Comment by kkimdev** [comment]
Hi, thanks for the fix!  Can this be back-ported to 1.16?  This is a blocking issue for us to use istio k8s gateway api.

```
2023-01-07T07:25:17.144609Z	warning	envoy config	StreamAggregatedResources gRPC config stream to xds-grpc closed: 3, failed to unmarshal node metadata ({"ANNOTATIONS":{"cert-manager.io/cluster-issuer":"selfsigned-cluster-issuer","inject.istio.io/templates":"gateway","kubernetes.io/config.seen":"2023-01-07T07:25:11.673825918Z","kubernetes.io/config.source":"api","prometheus.io/path":"/stats/prometheus","prometheus.io/port":"15020","prometheus.io/scrape":"true","proxy.istio.io/overrides":"{\"containers\":[{\"name\":\"istio-proxy\",\"ports\":[{\"name\":\"status-port\",\"containerPort\":15021,\"protocol\":\"TCP\"}],\"resources\":{},\"volumeMounts\":[{\"name\":\"kube-api-access-5bprz\",\"readOnly\":true,\"mountPath\":\"/var/run/secrets/kubernetes.io/serviceaccount\"}],\"readinessProbe\":{\"httpGet\":{\"path\":\"/healthz/ready\",\"port\":15021,\"scheme\":\"HTTP\"},\"timeoutSeconds\":2,\"periodSeconds\":2,\"successThreshold\":1,\"failureThreshold\":10},\"terminationMessagePath\":\"/dev/termination-log\",\"terminationMessagePolicy\":\"File\",\"imagePullPolicy\":\"Always\",\"securityContext\":{\"capabilities\":{\"drop\":[\"ALL\"]},\"privileged\":false,\"runAsUser\":1337,\"runAsGroup\":1337,\"runAsNonRoot\":true,\"readOnlyRootFilesystem\":true,\"allowPrivilegeEscalation\":false}}]}","sidecar.istio.io/status":"{\"initContainers\":null,\"containers\":[\"istio-proxy\"],\"volumes\":[\"workload-socket\",\"credential-socket\",\"workload-certs\",\"istio-envoy\",\"istio-data\",\"istio-podinfo\",\"istio-token\",\"istiod-ca-cert\"],\"imagePullSecrets\":null,\"revision\":\"default\"}"},"APP_CONTAINERS":"","CLUSTER_ID":"Kubernetes","ENVOY_PROMETHEUS_PORT":15090,"ENVOY_STATUS_PORT":15021,"INSTANCE_IPS":"10.42.0.62","INTERCEPTION_MODE":"REDIRECT","ISTIO_PROXY_SHA":"db8a88da7a0a3b8259a2f6b6ee4a806c23795a9f","ISTIO_VERSION":"1.16.1","LABELS":{"app.kubernetes.io/instance":"ppluto-main","istio.io/gateway-name":"ppluto-main-gateway","istio.io/rev":"default","service.istio.io/canonical-name":"ppluto-main-gateway","service.istio.io/canonical-revision":"latest","sidecar.istio.io/inject":"true"},"MESH_ID":"cluster.local","NAME":"ppluto-main-gateway-6cbbd7764d-4k475","NAMESPACE":"argocd","OWNER":"kubernetes://apis/apps/v1/namespaces/argocd/deployments/ppluto-main-gateway","PILOT_SAN":["istiod.istio-system.svc"],"POD_PORTS":"[{\"name\":\"status-port\",\"containerPort\":15021,\"protocol\":\"TCP\"}]","PROXY_CONFIG":{"binaryPath":"/usr/local/bin/envoy","configPath":"./etc/istio/proxy","controlPlaneAuthPolicy":"MUTUAL_TLS","discoveryAddress":"istiod.istio-system.svc:15012","drainDuration":"45s","parentShutdownDuration":"60s","proxyAdminPort":15000,"serviceCluster":"istio-proxy","statNameLength":189,"statusPort":15020,"terminationDrainDuration":"5s","tracing":{"zipkin":{"address":"zipkin.istio-system:9411"}}},"SERVICE_ACCOUNT":"default","WORKLOAD_NAME":"ppluto-main-gateway"}): unknown field "parentShutdownDuration" in istio.mesh.v1alpha1.ProxyConfig
```

**Comment by ramaraochavali** [comment]
@kkimdev This should not be there in 1.16 though?

**Pull request merged**